lib/src/web_socket/common/w_socket.dart

```diff
@@ -28,74 +28,51 @@ abstract class CommonWSocket extends Stream implements WSocket {
   /// no close reason available this property will be `null`.
   String closeReason;
 
-  /// Incoming communication. Data from the web socket will be piped
-  /// to this controller's sink. The stream will be exposed, allowing
-  /// users to listen to the incoming data.
-  StreamController incoming;
+  bool isClosed = false;
 
-  /// Outgoing communication. Sink will be exposed, allowing users to
-  /// add items to the outgoing stream.
-  StreamController outgoing;
+  StreamSubscription webSocketSubscription;
 
-  /// Completer that will complete when the outgoing sink has finished closing
-  /// and the socket connection has finished closing.
   Completer<Null> _allClosed = new Completer();
-
-  /// The close event with close code and reason from the socket connection.
-  /// Will only be populated once the socket has actually closed.
-  WSocketCloseEvent _closeEvent;
-
-  /// Completer that will complete when [_allClosed] has completed and the close
-  /// code and reason have been set. If the socket closed due to an error, this
-  /// will complete with an error. Otherwise, it will complete normally.
   Completer<Null> _done = new Completer();
-
-  /// The error that was either added to the socket sink or received from the
-  /// socket stream. This will only be populated if an error occurs.
   var _error;
-
-  /// Whether or not the web socket is in the process of closing (or already has
-  /// closed). This prevents duplicate behavior if [close] is called multiple
-  /// times.
-  bool _isClosed = false;
-
-  /// Whether or not the outgoing sink has been closed.
-  bool _isSinkClosed = false;
-
-  /// Whether or not the incoming socket stream connection has been closed.
-  bool _isSocketClosed = false;
-
-  /// The stack trace from the error that was either added to the socket sink or
-  /// received from the socket stream. This will only be populated if an error
-  /// occurs.
+  StreamController _incoming;
+  bool _isIncomingClosed = false;
+  bool _isOutgoingClosed = false;
+  StreamController _outgoing;
   StackTrace _stackTrace;
+  WSocketSubscription _incomingSubscription;
```

Comment:
I'll go back and add some docs for these. I also noticed I removed some docstrings that I didn't mean to, so I'll add those back.